packages/paste-core/components/combobox/src/index.tsx

```diff
@@ -0,0 +1,147 @@
+import * as React from 'react';
+import * as PropTypes from 'prop-types';
+import {useUID} from 'react-uid';
+import {useComboboxPrimitive} from '@twilio-paste/combobox-primitive';
+import {ChevronDownIcon} from '@twilio-paste/icons/esm/ChevronDownIcon';
+import {Box} from '@twilio-paste/box';
+import {FormHelpText, FormInput, FormLabel} from '@twilio-paste/form';
+import {Text} from '@twilio-paste/text';
+
+export interface ComboboxProps {
+  autocomplete?: boolean;
+  disabled?: boolean;
+  hasError?: boolean;
+  helpText?: string | React.ReactNode;
+  id?: string;
+  items: string[];
+  labelText: string | NonNullable<React.ReactNode>;
+  name?: string;
+  placeholder?: string;
+  required?: boolean;
+}
+
+const ComboboxIconWrapper: React.FC<{children: React.ReactNode}> = ({children}) => (
+  <Box
+    alignItems="center"
+    display="inline-flex"
+    position="absolute"
+    pointerEvents="none"
+    right="space40"
+    top="50%"
+    transform="translateY(-50%)"
+    zIndex="zIndex10"
+  >
+    {children}
+  </Box>
+);
+
+/* eslint-disable react/no-array-index-key */
+const Combobox: React.FC<ComboboxProps> = ({
+  autocomplete,
+  disabled,
+  helpText,
+  hasError,
+  id,
+  items,
+  labelText,
+  name,
+  placeholder,
+  required,
+}) => {
+  const [inputItems, setInputItems] = React.useState(items);
+  const {
+    getComboboxProps,
+    getInputProps,
+    getItemProps,
+    getLabelProps,
+    getMenuProps,
+    getToggleButtonProps,
+    highlightedIndex,
+    isOpen,
+  } = useComboboxPrimitive({
+    items: inputItems,
+    onInputValueChange: ({inputValue}) => {
+      if (autocomplete && inputValue) {
+        setInputItems(items.filter(item => item.toLowerCase().startsWith(inputValue.toLowerCase())));
+      }
+      return inputValue;
+    },
+  });
+  const helpTextId = useUID();
+  return (
+    <>
+      <FormLabel disabled={disabled} htmlFor={id} required={required} {...getLabelProps()}>
+        {labelText}
+      </FormLabel>
+      <Box {...getComboboxProps({role: 'combobox'})} display="flex" position="relative">
+        <FormInput
+          {...getInputProps()}
+          {...getToggleButtonProps({tabIndex: 0})}
```

Comment:
For the non-autocomplete combobox, I shouldn't be able to type into the input at all. Only Select something from the list. if you add an `onInput` handler to the input, and prevent the default behaviour of that event if it's not autocomplete, that _should_ do the trick. Ask if you don't know what those words mean.

Reply:
Worked this out using `selectedItem` from the primitive.

Reply:
I actually had to change this a bit for an instance when a user doesn't want autocomplete, and has a complete item object. `selectedItem` only handles strings, so it was returning `object Object`. Using `{...(!autocomplete ? {onChange: event => event.preventDefault()} : undefined)}` solves the issue.